src/host/readDataCooked.cpp

```diff
@@ -1028,6 +1028,8 @@ void COOKED_READ_DATA::SavePendingInput(const size_t index, const bool multiline
                 CONSOLE_INFORMATION& gci = ServiceLocator::LocateGlobals().getConsoleInformation();
                 LOG_IF_FAILED(_commandHistory->Add({ _backupLimit, StringLength / sizeof(wchar_t) },
                                                    WI_IsFlagSet(gci.Flags, CONSOLE_HISTORY_NODUP)));
+
+                Tracing::s_TraceCookedRead(_backupLimit);

```

Comment:
If there are too many client processes attached, some of them will not have a history list. We may want to place this outside the `if (_commandHistory)` block.

Reply:
Ah. Didn't realise that could be a thing. Makes sense will address.